providers/sparkpost/src/lib/sparkpost.provider.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  IEmailOptions,
+  IEmailProvider,
+} from '@novu/stateless';
+import { randomUUID } from 'crypto';
+import SparkPost from 'sparkpost';
+
+export class SparkPostEmailProvider implements IEmailProvider {
+  readonly id = 'sparkpost';
+  readonly channelType = ChannelTypeEnum.EMAIL as ChannelTypeEnum.EMAIL;
+  private readonly client: SparkPost;
+
+  constructor(apiKey: string) {
+    this.client = new SparkPost(apiKey);
```

Comment:
Sorry if this is an impolite review. 

I was looking up the spec, and it seems there should be an option here to configure the endpoint, mainly for GDPR reasons. EU and the rest of the world have different endpoints.

See: https://developers.sparkpost.com/api/#header-endpoints

For Node.js, you can do this with `options.endpoint` as mentioned here: https://github.com/SparkPost/node-sparkpost#initialization

I believe this can be made part of the constructor if you want.